### PR TITLE
Added ability to supply additional information with a Google Analytics

### PR DESCRIPTION
**Pull request by DmitryKlymenkoAU**
Hi,

First of all, thank you very much for this simple and beautifully working Google Analytics Events wrapper. What I was missing is an ability to supply additional (customer) payload to the Google Analytics Hit. In my case, I was tracking events to the Google Analytics View which does filter inbound traffic using hostnames matching specific format and since I wasn't able to supply Document Location or Page Path together with the hit things were off in my GA. By having this very simple and optional pass-thru for Google Analytics Measurement Protocol parameters/fields from caller code to the internal hit payload builder people will be able to supply additional pieces of data should they need.

Alternatively, I believe it would be beneficial to add support for screenname field, document path, document title and location fields.

Thank you very much for considering my PR.

### CODE REVIEW

File: Source/UniversalAnalyticsHttpWrapper/EventTracker.cs [hunk 1]
```diff
@@ -83,5 +86,31 @@ public async Task<TrackingResult> TrackEventAsync(IUniversalAnalyticsEvent analy
...
         }
+
+        /// <summary>
+        /// Adds "raw" (custom) payload to the hit data.
+        /// See https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters for parameters list. For example AddToCustomPayload("aip", "1") to enable IP anonymization.
+        /// If parameter was already added, it's value will be replaced with the supplied one
+        /// </summary>
+        /// <param name="name">Google Analytics Measurement Protocol short parameter name. for ex: aip, ds, qt, etc</param>
+        /// <param name="value">Parameter value</param>
+        public /*bool*/ void AddToCustomPayload(string name, string value)
```
Comment (jakejgordon) [inline Source/UniversalAnalyticsHttpWrapper/EventTracker.cs]: I think I prefer the ArgumentException in this case. If someone is developing against this they'll find out pretty quickly if they are setting a field they shouldn't this way :)

**Comment by jakejgordon** [comment]
Thanks for the great contribution, @DmitryKlymenkoAU ! Would you mind adding an example to the readme so people will know how to take advantage of the new feature more easily? Nice work :)

If you can get that change (and maybe remove the comments for returning bool instead of void in the one spot) I'll try to get this pushed up to NuGet within a day or two.

**Comment by DmitryKlymenkoAU** [comment]
Hi Jake,

Thank you very much for the reply.
I have applied changes you have requested and created an example in
readme.md file

Thanks
Dmitry

**Comment by jakejgordon** [comment]
Hi @DmitryKlymenkoAU I pulled the changes down locally and noticed that the unit tests fail on the EventTrackerTests (per this screenshot):

image

I also noticed that you didn't add unit tests for your changes. Is this something you're able to do? If possible, I'd love to keep some good test coverage here. Once again, I really appreciate the pull request and I'm looking forward to getting these changes deployed!

**Comment by DmitryKlymenkoAU** [comment]
Hi Jake,

No worries, at all. I have added test and made sure existing one still
works. Looks like Rhino has no support for parameters with default values :)
Reason I haven't done it in the first place - I had some troubles running
unit tests in my VS - I do not work in developer role and not too familiar
with these things.

If you are happy with it, please put it into the nuget package. If you
would like to add another couple of unit tests to cover edge cases of new
functionality, I am sure you will be able to do it much better than me.

Thanks and have a nice weekend
dmitry
